pkg/ebpf/common.go

```diff
@@ -78,27 +79,52 @@ func verifyOSVersion(kernelCode uint32, platform string, exclusionList []string)
 		return true, nil
 	}
 
-	if isCentOS(platform) && kernelCode >= centOSMinRequiredKernelCode {
-		return true, nil
-	}
-
 	if isUbuntu(platform) {
 		if kernelCode >= linuxKernelVersionCode(4, 4, 119) && kernelCode <= linuxKernelVersionCode(4, 4, 126) {
 			return false, fmt.Errorf("got ubuntu kernel %s with known bug on platform: %s, see: https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1763454", kernelCodeToString(kernelCode), platform)
 		}
 	}
 
-	if kernelCode < minRequiredKernelCode {
-		return false, fmt.Errorf(
-			"incompatible linux version. at least %s (%d) required, got %s (%d)",
-			kernelCodeToString(minRequiredKernelCode),
-			minRequiredKernelCode,
-			kernelCodeToString(kernelCode),
-			kernelCode,
-		)
+	kallsyms, err := readKallsyms()
+	if err != nil {
+		log.Warnf("error reading /proc/kallsyms file: %s", err)
+		// If we can't read the /proc/kallsyms file let's just return true to avoid blocking the tracer from running
+		return true, nil
+	}
+
+	return verifyKernelFuncs(kallsyms), nil
+}
+
+func readKallsyms() (string, error) {
+	procRoot := util.GetProcRoot()
+	raw, err := ioutil.ReadFile(path.Join(procRoot, "kallsyms"))
+	if err != nil {
+		return "", errors.Wrapf(err, "error reading kallsyms file from proc dir: %s", procRoot)
 	}
 
-	return true, nil
+	return string(raw), nil
+}
+
+func verifyKernelFuncs(kallsyms string) bool {
+	lines := strings.Split(kallsyms, "\n")
+	funcs := make(map[string]struct{}, len(lines))
```

Comment:
Can we write this without allocating such a large map?

I think something like:

```
found := map()
for _, requiredFunc := range required {
    found[requiredFunc] = false
}

for _, l := range lines {
   func := getFunc(l)
   if _, ok : =found[func] { found[func] = true }
}

// return true iff all entries in found are true
```